pandas/core/generic.py

```diff
@@ -4331,9 +4331,14 @@ def fillna(self, value=None, method=None, axis=None, inplace=False,
                 return result if not inplace else None
 
             elif not is_list_like(value):
-                new_data = self._data.fillna(value=value, limit=limit,
-                                             inplace=inplace,
-                                             downcast=downcast)
+                if axis == 0:
+                    target = self._data
+                elif axis == 1:
+                    target = self.T
+                new_data = target.fillna(value=value, limit=limit,
```

Comment:
see my comment above, I would push this *into* the internal implementation of fillna. (which can do the transposing internally), you simply need to add an axis parameter and pass as (these are passed inverted FYI).

```
                                 axis=self._get_block_manager_axis(axis),
 ```